ark/jsonschema/scope.ts

```diff
@@ -0,0 +1,108 @@
+import type { JsonSchemaOrBoolean } from "@ark/schema"
+import { type JsonSchema, scope, type Scope } from "arktype"
+
+type AnyKeywords = Partial<JsonSchema.Const & JsonSchema.Enum>
+
+type TypeWithNoKeywords = { type: "boolean" | "null" }
+type TypeWithKeywords =
+	| JsonSchema.Array
+	| JsonSchema.Numeric
+	| JsonSchema.Object
+	| StringSchema
+// NB: For sake of simplicitly, at runtime it's assumed that
+// whatever we're parsing is valid JSON since it will be 99% of the time.
+// This decision may be changed later, e.g. when a built-in JSON type exists in AT.
+type Json = unknown
+
+type ArraySchema = JsonSchema.Array
+
+type NumberSchema = JsonSchema.Numeric
+
+type ObjectSchema = JsonSchema.Object
+
+// NB: @ark/jsonschema doesn't support the "format" keyword, and the "pattern" could be string|RegExp rather than only string, so we need a separate type
```

Comment:
Hopefully we'll be able to align these with upcoming work on JSON schema but this looks like good for now